File: output/gpt_4o/avatar/algo_based_translation/python/java/successful_translation/codeforces_583_B.java

import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class codeforces_583_B {
    public static void main(String[] args) {
        mainFunction();
    }

    public static void mainFunction() {
        Scanner scanner = new Scanner(System.in);
        
        int n = Integer.parseInt(scanner.nextLine());
        String[] inputArray = scanner.nextLine().split(" ");
        List<Integer> a = new ArrayList<>();
        
        for (String s : inputArray) {
            a.add(Integer.parseInt(s));
        }
        
        int currentLevel = 0;
        int counter = 0;
        int counterTurns = -1;
        
        while (counter < a.size()) {
            counterTurns++;
            for (int i = 0; i < a.size(); i++) {
                if (a.get(i) != -1 && a.get(i) <= counter) {
                    counter++;
                    a.set(i, -1);
                }
            }
            Collections.reverse(a);
        }
        
        System.out.println(counterTurns);
    }
}

File: dataset/avatar/Python/Code/codeforces_583_B.py
def main_function ( ) :
    n = int ( input ( ) )
    a = [ int ( i ) for i in input ( ).split ( " " ) ]
    current_level = 0
    counter = 0
    counter_turns = - 1
    while counter < len ( a ) :
        counter_turns += 1
        for i in range ( len ( a ) ) :
            if a [ i ] != - 1 and a [ i ] <= counter :
                counter += 1
                a [ i ] = - 1
        a.reverse ( )
    print ( counter_turns )
main_function ( )

